src/basic/handlers/basic.ts

```diff
@@ -172,19 +174,32 @@ class BasicStepHandler implements FRUIStepHandler {
       case CallbackType.DeviceProfileCallback:
         return new DeviceProfileCallbackRenderer(cb as DeviceProfileCallback, this.onChange);
 
+      case CallbackType.KbaCreateCallback:
+        const kbaCreateCallback = cb as KbaCreateCallback;
+        const kbaIndex = this.step
+          .getCallbacksOfType<KbaCreateCallback>(CallbackType.KbaCreateCallback)
+          .filter((x) => x.getPrompt() === kbaCreateCallback.getPrompt())
+          .indexOf(kbaCreateCallback);
+        return new KbaCreateCallbackRenderer(kbaCreateCallback, index, kbaIndex, this.onChange);
+
       case CallbackType.PasswordCallback:
         return new PasswordCallbackRenderer(cb as PasswordCallback, index, this.onChange);
 
-      case CallbackType.ValidatedCreatePasswordCallback:
-        const passwordCallback = cb as ValidatedCreatePasswordCallback;
-        return new PasswordCallbackRenderer(passwordCallback, index, this.onChange);
-
       case CallbackType.PollingWaitCallback:
         return new PollingWaitCallbackRenderer(cb as PollingWaitCallback, index, this.onChange);
 
       case CallbackType.ReCaptchaCallback:
         return new ReCaptchaCallbackRenderer(cb as ReCaptchaCallback, index, this.onChange);
 
+      case CallbackType.SelectIdPCallback:
+        const hasLocalAuthForm = !!this.step.getCallbacksOfType(CallbackType.NameCallback).length;
+        return new SelectIdPCallbackRenderer(
+          cb as SelectIdPCallback,
+          index,
+          this.onChange,
+          hasLocalAuthForm,
+        );
```

Comment:
This selector is unique in that it needs to know if other local authentication input callbacks are present.